### PR TITLE
Added integration support for DataDog APM Tracing

### PR DESCRIPTION
**Pull request by aantono**
### What does this PR do?

Adds integration support for DataDog APM Tracing

### Motivation

Given that DataDog Tracing is OpenTracing API compliant, and Traefik already has support for DataDog monitoring, having ability to also send tracing would be beneficial

### CODE REVIEW

**Comment by mmatur** [comment]
Thanks @aantono for this PR, I will take a look ;)

File: docs/configuration/tracing.md [hunk 1]
```diff
@@ -98,3 +98,41 @@ Træfik supports two backends: Jaeger and Zipkin.
...
+```toml
+# Tracing definition
+[tracing]
+  # Backend name used to send tracing data
+  #
+  # Default: "jaeger"
+  #
+  backend = "datadog"
+
+  # Service name used in Zipkin backend
```
Comment (mmatur) [inline docs/configuration/tracing.md]: Could you replace `Service name used in Zipkin backend` by `Service name used in DataDog backend`

File: docs/configuration/tracing.md [hunk 2]
```diff
@@ -98,3 +98,41 @@ Træfik supports two backends: Jaeger and Zipkin.
...
+  #
+  # Default: "traefik"
+  #
+  serviceName = "traefik"
+
+  [tracing.datadog]
+    # Local Agent Host Port instructs reporter to send spans to datadog-tracing-agent at this address
+    #
+    # Default: "127.0.0.1:6831"
```
Comment (mmatur) [inline docs/configuration/tracing.md]: Could you replace `127.0.0.1:6831` by `127.0.0.1:8126`

File: docs/configuration/tracing.md [hunk 3]
```diff
@@ -98,3 +98,41 @@ Træfik supports two backends: Jaeger and Zipkin.
...
+    # Default: false
+    #
+    debug = false
+
+    # Apply shared tag in a form of Key:Value to all the traces
+    #
+    # Default: ""
+    #
+    globalTag = "foo:bar"
+
```
Comment (mmatur) [inline docs/configuration/tracing.md]: Could you remove this line please

File: middlewares/tracing/datadog/datadog.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	"io"
+
+	"github.com/containous/traefik/log"
+	"github.com/opentracing/opentracing-go"
+
+	ddtracer "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/opentracer"
+	datadog "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"strings"
```
Comment (mmatur) [inline middlewares/tracing/datadog/datadog.go]: Could you reorganize your import please?

File: middlewares/tracing/datadog/datadog.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+)
+
+// Name sets the name of this tracer
+const Name = "datadog"
+
+// Config provides configuration settings for a zipkin tracer
+type Config struct {
+	LocalAgentHostPort string `description:"set datadog-agent's host:port that the reporter will used. Defaults to localhost:8126" export:"false"`
```
Comment (mmatur) [inline middlewares/tracing/datadog/datadog.go]: Could you please replace `set` by `Set`

File: middlewares/tracing/datadog/datadog.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+	GlobalTag          string `description:"Key:Value tag to be set on all the spans." export:"true"`
+	Debug              bool   `description:"Enable Zipkin debug." export:"true"`
```
Comment (mmatur) [inline middlewares/tracing/datadog/datadog.go]: Could you replace `Zipkin` by `DataDog`?

File: middlewares/tracing/datadog/datadog.go [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/opentracing/opentracing-go"
+
+	ddtracer "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/opentracer"
+	datadog "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Name sets the name of this tracer
+const Name = "datadog"
+
+// Config provides configuration settings for a zipkin tracer
```
Comment (mmatur) [inline middlewares/tracing/datadog/datadog.go]: Could you change `zipkin` with `datadog` please

File: middlewares/tracing/datadog/datadog.go [hunk 8]
```diff
@@ -0,0 +1,46 @@
+package datadog
+
+import (
+	"io"
+	"strings"
+
+	"github.com/containous/traefik/log"
+	"github.com/opentracing/opentracing-go"
+
```
Comment (mmatur) [inline middlewares/tracing/datadog/datadog.go]: Could you please remove this line

Comment (aantono) [reply middlewares/tracing/datadog/datadog.go]: Done

**Pull request merged**